python/hawkey/subject-py.cpp

```diff
@@ -89,15 +89,21 @@ subject_dealloc(_SubjectObject *self)
 static int
 subject_init(_SubjectObject *self, PyObject *args, PyObject *kwds)
 {
-    const char * pattern;
+    PyObject *py_pattern = NULL;
     PyObject *icase = NULL;
     const char *kwlist[] = { "pattern", "ignore_case", NULL };
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "s|O!", (char**) kwlist, &pattern,
-        &PyBool_Type, &icase)) {
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|O!", (char**) kwlist,
+        &py_pattern, &PyBool_Type, &icase)) {
         return -1;
     }
-    self->pattern = g_strdup(pattern);
     self->icase = icase != NULL && PyObject_IsTrue(icase);
+    PyObject *tmp_py_str = NULL;
+    const char * pattern = pycomp_get_string(py_pattern, &tmp_py_str);
+    if (!pattern)
+        Py_XDECREF(tmp_py_str);
```

Comment:
`-Werror=misleading-indentation` would fail on this; see https://developers.redhat.com/blog/2016/02/26/gcc-6-wmisleading-indentation-vs-goto-fail/ and note in libostree we [add it to our CFLAGS by default](https://github.com/ostreedev/ostree/blob/master/configure.ac#L49), it'd be a good idea to do the same here if someone knew the relevant cmake incantations.

Reply:
Oh, that's stupid mistake coming from my strong python programming habits. Fixed.